pkg/kubelet/util/manager/cache_based_manager.go

```diff
@@ -144,28 +146,25 @@ func GetObjectTTLFromNodeFunc(getNode func() (*v1.Node, error)) GetObjectTTLFunc
 	}
 }
 
-func (s *objectStore) isObjectFresh(data *objectData) bool {
+func (s *objectStore) isObjectFresh(data *objectData, refUpdated bool) bool {
 	objectTTL := s.defaultTTL
 	if ttl, ok := s.getTTL(); ok {
 		objectTTL = ttl
 	}
-	return s.clock.Now().Before(data.lastUpdateTime.Add(objectTTL))
+	return !refUpdated && s.clock.Now().Before(data.lastUpdateTime.Add(objectTTL))
```

Comment:
if `refUpdated`, you can just shortcircuit. That seems like a better check to inline where the function is called though.

Reply:
I meant check refUpdated where you call `isObjectFresh` rather than passing it in.